lib/temporal/testing/replay_tester.rb

```diff
@@ -0,0 +1,136 @@
+      def replay_history_protobuf(workflow_class, proto)
+        raw_history = Temporalio::Api::History::V1::History.decode(proto)
+        replay_history(workflow_class, Workflow::History.new(raw_history.events))
+      end
+
+      # Runs a replay test by using the specific Temporal::Workflow::History object. This can only be obtained by
+      # calling the .get_workflow_history method on the Temporal client to download a history from Temporal server
+      # into memory.
+      #
+      # If the replay test succeeds, the method will return silently. If the replay tests fails, an error will be raised.
+      def replay_history(workflow_class, history)
+        # This code roughly resembles the workflow TaskProcessor but with history being fed in rather
+        # than being pulled via a workflow task, no query support, no metrics, and other
+        # simplifications. Fake metadata needs to be provided.
+        metadata = Temporal::Metadata::WorkflowTask.new(
+          namespace: 'replay-test',
+          id: 1,
+          task_token: '',
+          attempt: 1,
+          workflow_run_id: 'run_id',
+          workflow_id: 'workflow_id',
+          workflow_name: history.find_event_by_id(1).attributes.workflow_type.name
+        )
+
+        executor = Workflow::Executor.new(
+          workflow_class,
+          history,
+          metadata,
+          @config,
+          true,
+          Middleware::Chain.new([])
+        )
+
+        begin
+          executor.run
+        rescue StandardError
+          query = Struct.new(:query_type, :query_args).new(
+            Temporal::Workflow::StackTraceTracker::STACK_TRACE_QUERY_NAME,
+            nil
+          )
+          query_result = executor.process_queries(
+            { 'stack_trace' => query }
+          )
+          replay_error = ReplayError.new('Workflow code failed to replay successfully against history')
+          # Override the stack trace to the point in the workflow code where the failure occured, not the
+          # point in the StateManager where non-determinism is detected
+          replay_error.set_backtrace("Fiber backtraces: #{query_result['stack_trace'].result}")
+          raise replay_error
+        end
+      end
+
+      # Protobuf does not define a consistent format for enums in JSON. In some Temporal SDKs, the event
+      # type enum takes the form 'WorkflowTaskScheduled' but here it needs to be EVENT_TYPE_WORKFLOW_TASK_SCHEDULED.
+      # This code parses the JSON, reformats these fields, then regenerates the JSON. This should not be necessary
+      # for histories downloaded from recent versions of the Temporal CLI, Temporal UI, or using methods
+      # on Temporal::Client.
+      #
+      # If the pretty_print optional parameter is set to true, it outputs in a more human
+      # readable form on output.
+      def self.correct_event_types(text, pretty_print: true)
```

Comment:
Is this because Stripe is on an old version of the Java SDK?  Don't see this function used anywhere.  Should it be tested?  Or maybe just done on the stripe fork?

Reply:
I believe this originally started with older versions of tctl. Unfortunately, we standardized our internal tooling on this camel case enum style rather than the standard screaming snake case version. It probably does make more sense to leave it out of here, as any new histories collected using current Temporal UI/CLI or these new APIs won't need to rely on this. I agree if it's kept, it should be tested.